src/librustc/ty/print/pretty.rs

```diff
@@ -896,32 +913,49 @@ pub trait PrettyPrinter<'tcx>:
         Ok(self)
     }
 
-    fn pretty_print_const_value(
+    fn pretty_print_const_scalar(
         mut self,
-        ct: ConstValue<'tcx>,
+        scalar: Scalar,
         ty: Ty<'tcx>,
         print_ty: bool,
     ) -> Result<Self::Const, Self::Error> {
         define_scoped_cx!(self);
 
-        if self.tcx().sess.verbose() {
-            p!(write("ConstValue({:?}: {:?})", ct, ty));
-            return Ok(self);
-        }
-
-        let u8 = self.tcx().types.u8;
-
-        match (ct, &ty.kind) {
-            (ConstValue::Scalar(Scalar::Raw { data, .. }), ty::Bool) => {
-                p!(write("{}", if data == 0 { "false" } else { "true" }))
-            }
-            (ConstValue::Scalar(Scalar::Raw { data, .. }), ty::Float(ast::FloatTy::F32)) => {
+        match (scalar, &ty.kind) {
+            // Single element arrays print their element (they are `#[transparent]`) enclosed in
+            // square brackets.
+            (_, ty::Array(t, n)) if n.eval_usize(self.tcx(), ty::ParamEnv::empty()) == 1 => {
+                p!(write("["));
+                self = self.pretty_print_const_scalar(scalar, t, print_ty)?;
+                p!(write("]"));
+            }
```

Comment:
Do we need this special-case? Seems unnecessary to me, but also I think it should be handled uniformly by some kind of ADT traversal.

Reply:
The reason this exists is that we don't have a generic ADT traversal yet but in this simple case we can just print the single element. I can remove it though, single element array are rare anyway

Reply:
I removed it